Upgrade to Monito 1.7.1 Breaks the UI Tests

### Description
Upgrading to the latest [Monitor](https://developer.android.com/jetpack/androidx/releases/test#monitor-1.7.1) from `v1.6.1` to `v1.7.1` are breaking the UI tests

### Expected Results
The UI tests should work as expected

### Actual Results
The UI tests are breaking with the stack trace:
`Failed resolution of: Landroidx/test/internal/platform/app/ActivityInvoker$-CC; at androidx.test.core.app.InstrumentationActivityInvoker.getIntentForActivity(Unknown Source:0) at androidx.test.core.app.InstrumentationActivityInvoker.startEmptyActivitySync(InstrumentationActivityInvoker.java:547) at androidx.test.core.app.InstrumentationActivityInvoker.finishActivity(InstrumentationActivityInvoker.java:585) at androidx.test.core.app.ActivityScenario.moveToState(ActivityScenario.java:677) at androidx.test.core.app.ActivityScenario.close(ActivityScenario.java:416) at androidx.test.ext.junit.rules.ActivityScenarioRule.after(ActivityScenarioRule.java:115) at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:59) at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100) at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366) at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103) at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63) at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) at org.junit.runners.ParentRunner.run(ParentRunner.java:413) at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:162) at org.junit.runners.Suite.runChild(Suite.java:128) at org.junit.runners.Suite.runChild(Suite.java:27) at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) at org.junit.runners.ParentRunner.run(ParentRunner.java:413) at org.junit.runner.JUnitCore.run(JUnitCore.java:137) at org.junit.runner.JUnitCore.run(JUnitCore.java:115) at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:67) at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:58) at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:446) at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2205) Caused by: java.lang.ClassNotFoundException: Didn't find class "androidx.test.internal.platform.app.ActivityInvoker$-CC" on path: DexPathList[[zip file "/system/framework/android.test.runner.jar", zip file "/system/framework/android.test.mock.jar", zip file "/system/framework/android.test.base.jar", zip file "/data/app/~~9SWORgRdLFDx7TVLA395rw==/uk.co.bbc.android.sport.testing.endtoend-SbQNCQaeCW2fQDkIzacEYQ==/base.apk", zip file "/data/app/~~LUtmJvvuH344hpESBKIVNA==/uk.co.bbc.android.sportdomestic.debug-EVJ1mQrTpiSGAjxXnTX-pA==/base.apk"],nativeLibraryDirectories=[/data/app/~~9SWORgRdLFDx7TVLA395rw==/uk.co.bbc.android.sport.testing.endtoend-SbQNCQaeCW2fQDkIzacEYQ==/lib/arm64, /data/app/~~LUtmJvvuH344hpESBKIVNA==/uk.co.bbc.android.sportdomestic.debug-EVJ1mQrTpiSGAjxXnTX-pA==/lib/arm64, /system/lib64, /system_ext/lib64]] at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:207) at java.lang.ClassLoader.loadClass(ClassLoader.java:379) at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
`

### AndroidX Test and Android OS Versions
   - compileSdk = 34
   - android_test_junit_lib = "androidx.test.ext:junit:1.2.1"
   - android_test_runner_lib = "androidx.test:runner:1.5.2"
   - android_test_orchestrator_lib = "androidx.test:orchestrator:1.4.2"
   - android_test_truth_lib = "androidx.test.ext:truth:1.5.0"
   - androidx_core_testing_lib = "androidx.arch.core:core-testing:2.2.0"
   - androidx_monitor_testing_lib = "androidx.test:monitor:1.6.1"
   - espresso_contrib_lib = "androidx.test.espresso:espresso-contrib:3.6.1"
   - espresso_intents_lib = "androidx.test.espresso:espresso-intents:3.6.1"
   - espresso_lib = "androidx.test.espresso:espresso-core:3.6.1"
   - espresso_accessibility_lib = "androidx.test.espresso:espresso-accessibility:3.6.1"
   - fragment_testing_lib = "androidx.fragment:fragment-testing:1.8.1"

--- a/runner/monitor/java/androidx/test/BUILD
+++ b/runner/monitor/java/androidx/test/BUILD
@@ -35,6 +35,7 @@ kt_android_library(
             "internal/runner/runtime/ExposedInstrumentationApi.java",
             "internal/runner/hidden/ExposedInstrumentationApi.java",
             "internal/platform/app/ActivityInvoker$$CC.java",
+            "internal/platform/app/ActivityInvokerDesugar.java",
         ],
     ),
     tags = ["alt_dep=//runner/monitor"],
@@ -81,6 +82,7 @@ android_library(
     srcs = [
         # only needed for external release backwards compatibility
         "internal/platform/app/ActivityInvoker$$CC.java",
+        "internal/platform/app/ActivityInvokerDesugar.java",
     ],
     custom_package = "androidx.test.monitor",
     manifest = "AndroidManifest.xml",
@@ -107,6 +109,7 @@ axt_android_aar(
         "androidx.test.platform",
         "androidx.test.runner",
     ],
+    jarjar_rule = ":jarjar.txt",
     included_dep = ":monitor_release_lib",
 )
 

--- a/runner/monitor/java/androidx/test/internal/platform/app/ActivityInvokerDesugar.java
+++ b/runner/monitor/java/androidx/test/internal/platform/app/ActivityInvokerDesugar.java
@@ -1,0 +1,40 @@
+package androidx.test.internal.platform.app;
+
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+
+import android.app.Activity;
+import android.content.ComponentName;
+import android.content.Intent;
+import androidx.annotation.RestrictTo;
+import androidx.annotation.RestrictTo.Scope;
+
+/**
+ * Handles default implementation for ActivityInvoker#getIntentForActivity
+ *
+ * <p> See the {@link ActivityInvoker$$CC} javadoc for prior history.
+ *
+ * <p> Starting in androidx.test:monitor:1.6.X, a new version of the desugar tooling was used
+ * that generated a ActivityInvoker$-CC class. Class names with hyphens are rejected by javac,
+ * so we cannot directly declare a ActivityInvoker$-CC class here. So instead we use a
+ * placeholder name, and use jarjar to rename the class after the javac step.
+ *
+ * @hide
+ */
+@RestrictTo(Scope.LIBRARY_GROUP)
+public final class ActivityInvokerDesugar {
+
+  private ActivityInvokerDesugar() {}
+
+  public static Intent $default$getIntentForActivity(ActivityInvoker invoker,
+            Class<? extends Activity> activityClass) {
+    Intent intent =
+        Intent.makeMainActivity(
+            new ComponentName(getInstrumentation().getTargetContext(), activityClass));
+    if (getInstrumentation().getTargetContext().getPackageManager().resolveActivity(intent, 0)
+        != null) {
+      return intent;
+    }
+    return Intent.makeMainActivity(
+        new ComponentName(getInstrumentation().getContext(), activityClass));
+  }
+}

--- a/runner/monitor/java/androidx/test/jarjar.txt
+++ b/runner/monitor/java/androidx/test/jarjar.txt
@@ -1,0 +1,1 @@
+rule androidx.test.internal.platform.app.ActivityInvokerDesugar androidx.test.internal.platform.app.ActivityInvoker$-CC

